crates/sp-domains/src/fraud_proof.rs

```diff
@@ -414,12 +416,40 @@ pub struct InvalidTotalRewardsProof {
     pub storage_proof: StorageProof,
 }
 
+#[derive(Clone, Debug, Decode, Encode, Eq, PartialEq, TypeInfo)]
+pub enum ExtrinsicDigest {
+    /// Actual extrinsic data that is inlined since it is less than 33 bytes.
```

Comment:
No, StateVersion::V1 inlines the actual data as is if it is `<= 32` else it will hash the data and insert is as trie node. So we need to pass the extrinsic data as is if the extrinsic data needs to be inlined.